packages/phisherman/src/lib/Phisherman.ts

```diff
@@ -0,0 +1,82 @@
+import { fetch, FetchMethods, FetchResultTypes, QueryError } from '@sapphire/fetch';
+import type { PhishermanReportType, PhishermanReturnType } from './PhishermanTypes';
+
+let storedApiKey: string;
+
+/**
+ * Checks if a link is detected as a scam or phishing link by phisherman.
+ * @param domain The domain to check.
+ * @since 1.0.0
+ */
+export async function checkDomain(domain: string, apiKey: string = storedApiKey) {
+	const result = await fetch<PhishermanReturnType>(
+		`https://api.phisherman.gg/v2/domains/check/${domain}`,
+		{
+			headers: {
+				'Content-Type': 'application/json',
+				Authorization: `Bearer ${apiKey}`
+			}
```

Comment:
Add a User-Agent of our lib

Reply:
Specifically:
```
'User-Agent': 'Sapphire Phisherman/1.0.0 (node-fetch) ${os.platform()}/${os.release()}
```

Add to the top of the file:
```ts
import os from 'node:os';
```

Reply:
Edited in the link for @UndiedGamer